sorc/chgres_cube.fd/input_data.F90

```diff
@@ -6607,4 +6603,40 @@ recursive subroutine quicksort(a, first, last)
   if (j+1 < last)  call quicksort(a, j+1, last)
 end subroutine quicksort
 
+!> Check for and replace certain values in soil temperature.
+!> At open water points (landmask=0) use skin temperature as
+!> a filler value. At land points (landmask=1) with excessive
+!> soil temperature, replace soil temperature with skin temperature. 
+!> In GEFSv12.0 data there are some erroneous missing values at
+!> land points which this corrects. At sea ice points (landmask=2),
+!> store a default ice column temperature because grib2 files do not 
+!> have ice column temperature which FV3 expects at these points.
+!!
+!! @param soilt    [inout] 3-dimensional soil temperature arrray
+!! @param landmask [in]    landmask of the input grid
+!! @param skint    [in]    2-dimensional skin temperature array
+!! @author Larissa Reames CIMMS/NSSL
+
+subroutine check_soilt(soilt, landmask, skint)
+  implicit none
+  real(esmf_kind_r8), intent(inout) ::  soilt(i_input,j_input,lsoil_input)
+  real(esmf_kind_r8), intent(in)    ::  skint(i_input,j_input)
+  integer(esmf_kind_i4), intent(in)    ::  landmask(i_input,j_input)
+  
+  integer                           :: i, j, k
+
+  do k=1,lsoil_input
+    do j = 1, j_input
+      do i = 1, i_input
+        if (landmask(i,j) == 0_esmf_kind_i4 ) then 
+          soilt(i,j,k) = skint(i,j)
+        else if (landmask(i,j) == 1_esmf_kind_i4 .and. soilt(i,j,k) > 350.0_esmf_kind_r8) then 
+          soilt(i,j,k) = skint(i,j)
+        else if (landmask(i,j) == 2_esmf_kind_i4 ) then 
+          soilt(i,j,k) = icet_default
```

Comment:
This has been fix. I'd like to request that we start keeping a wiki or something that we can choose to get updates about when these kinds of decisions are made. I imagine, going forward, they're going to become more common and it would be helpful to know ahead of time so we can make sure any new/modified code conforms to these choices.

Reply:
> This has been fixed. I'd like to request that we start keeping a wiki or something that we can choose to get updates about when these kinds of decisions are made. I imagine, going forward, they're going to become more common and it would be helpful to know ahead of time so we can make sure any new/modified code conforms to these choices.

Totally agree.